chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -628,47 +618,113 @@ impl<'a, S: TransactionVerifierStorageRef> TransactionVerifier<'a, S> {
         Ok(fee)
     }
 
+    pub fn can_disconnect_transaction(
+        &mut self,
+        tx_source: &TransactionSource,
+        tx_id: &Id<Transaction>,
+    ) -> bool {
+        match tx_source {
+            TransactionSource::Chain(block_id) => {
+                let current_block_height =
+                    match self.storage_ref.get_gen_block_index(&(*block_id).into()) {
+                        Ok(block_index) => match block_index {
+                            Some(block_index) => block_index.block_height(),
+                            None => return false,
```

Comment:
I don't like all these explicit `return false`, but I don't know how to ignore Err from Result in a cleaner way.
My intention was to treat errors in this function as negative result, so that the user could simply write:
```rust
if verifier.can_disconnect_transaction(tx) {
```

If someone has a suggestion, please comment